CODINGSTYLE.md

```diff
@@ -0,0 +1,335 @@
+# Coding Style Guide
+
+The Scala.js project has pretty drastic rules in terms of coding style.
+This document tries to document them as much as possible.
+
+A *few* of these rules are checked automatically using Scalastyle, but most of them are too complex to teach to an automated tool.
+
+In some cases, [@sjrd](https://github.com/sjrd) will just say something else on a code review.
+His opinion on a direct piece of code always takes precedence over this document.
+Feel free to point out the conflict, though: the document can be improved.
```

Comment:
just to be honest at a first read I definitely not appreciate that sentence too.
I perfectly understand that it is not indeed although It looks arrogant.

You can maybe say something that sounds like:
```our coding guidelines are strongly driven by @sjrd who is actually in charge to define and maintain them, please follow his direction....```